dotnetapp-buildcontainer/build.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+# Capture the current build and app image, if any, so that we can clean them up when we are done.
+currentBuildImage=$(docker images build-image -q)
+currentAppImage=$(docker images dockerapp -q)
+
+docker build -t build-image -f Dockerfile.build .
+
+# Create a container from our build image so that we can copy the contents out.
+docker create --name build-container build-image
+# Copy the contents of the /out directory out so that we can build our app image.
+docker cp build-container:/out ./out
+# Build the application image.
+docker build -t dockerapp -f ./out/Dockerfile ./out
```

Comment:
I'm not following how that is going to prevent the Dockerfile from being added to the dockerapp image.  I was thinking you would need to do something like, stop coping the Dockerfile to the `out` directory and build via `docker build -t dockerapp .` Lastly the Dockerfile would have to be modified to copy from `out` - https://github.com/dotnet/dotnet-docker-samples/commit/a74d917820c6e2e3c7ee8f8de408974f20f5c8ed